File: K8sDemoApp/src/app/_models/Hub_Messages/DirectorStatusMessage.ts
import { DateEditTemplateComponent } from "@generic-ui/ngx-grid/composition/core/domain-read/edit/template/date-edit-template.component";
import { JobTypeEnum } from "src/app/_enum/JobTypeEnum";

export interface DirectorStatusMessage{
    timestamp: Date,
    registeredWorkers: Array<WorkerDescriptor>,
    totalJobs:number,
    maxConcurrentTasks:number,
    maxWorkers:number,
    maxOpenJobDuration:number,
}

export interface WorkerDescriptor{
    workerId:string,
    currentJobs:number
}

File: k8sCore/Messages/DirectorStatusMessage.cs
using System;
using System.Collections.Generic;
using K8sCore.DTOs;

namespace K8sCore.Messages
{
    public class DirectorStatusMessage
    {

        public DateTime Timestamp { get; set; }
        public List<WorkerDescriptorDto> RegisteredWorkers { get; set; }
        public int TotalJobs { get; set; }
        public int MaxConcurrentTasks { get; set; }
        public int MaxWorkers { get; set; }
        
        
    }
}